core/src/test/java/org/apache/calcite/tools/PlannerTest.java

```diff
@@ -1525,4 +1527,65 @@ private void checkView(String sql, Matcher<String> matcher)
     final RelRoot root = planner.rel(validate);
     assertThat(toString(root.rel), matcher);
   }
+
+  /**
+   *  Helper function that checks that a Planner uses the correct type system when provided.
+   */
+  private void checkTypeSystem(String sql, RelDataTypeSystem typeSystem, String expectedRelExpr)
```

Comment:
Small notes:

1. The naming does not reflect the actual implementation.
The better name would be `assertValidatedPlan`.
2. It is better place the method _after_ the test method, so the code is easier to read from top to bottom (higher-level methods shoud come before low-level ones, especially when lower-level methods have just one usage)